StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/PaymentSheet.swift

```diff
@@ -153,21 +153,25 @@ public class PaymentSheet {
             switch result {
             case .success(let loadResult):
                 // Set the PaymentSheetViewController as the content of our bottom sheet
-                let presentPaymentSheetVC = {
-                    let paymentSheetVC = PaymentSheetViewController(
-                        configuration: self.configuration,
-                        loadResult: loadResult,
-                        delegate: self
-                    )
-
-                    self.configuration.style.configure(paymentSheetVC)
-
-                    let updateBottomSheet: () -> Void = {
-                        self.bottomSheetViewController.contentStack = [paymentSheetVC]
+                let paymentSheetVC: PaymentSheetViewControllerProtocol = {
+                    switch self.configuration.appearance.layout {
+                    case .horizontal:
+                        return PaymentSheetViewController(
+                            configuration: self.configuration,
+                            loadResult: loadResult,
+                            delegate: self
+                        )
+                    case .vertical:
+                        let verticalVC = PaymentSheetVerticalViewController(
+                            configuration: self.configuration,
+                            loadResult: loadResult,
+                            isFlowController: false
+                        )
+                        verticalVC.paymentSheetDelegate = self
+                        return verticalVC
                     }
-                    updateBottomSheet()
-                }
-                presentPaymentSheetVC()
+                }()
+                self.bottomSheetViewController.contentStack = [paymentSheetVC]
```

Comment:
Any reason we set the stack explicitly instead of using `pushContentViewController`?

Reply:
I think it's so that we can replace the loading screen.